controllers/ovndbcluster_controller.go

```diff
@@ -579,8 +571,117 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 			if err != nil {
 				return ctrl.Result{}, err
 			}
+			// Delete DNS records for deleted services/pods
+			dnsData := &infranetworkv1.DNSData{}
+			dnsName := fmt.Sprintf("dns-%s-%d", serviceName, i)
+			dnsNamespace := helper.GetBeforeObject().GetNamespace()
+			err = helper.GetClient().Get(ctx, types.NamespacedName{Name: dnsName, Namespace: dnsNamespace}, dnsData)
+			if err != nil && !k8s_errors.IsNotFound(err) {
+				Log.Info(fmt.Sprintf("Error while getting DNS object: %v", err))
+			}
+			err := helper.GetClient().Delete(ctx, dnsData)
+			if err != nil {
+				Log.Info(fmt.Sprintf("Error while cleaning the DNS record %s: %v", dnsName, err))
+			}
+
+		}
+	}
+
+	podList, err = ovndbcluster.OVNDBPods(ctx, instance, helper, serviceLabels)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	// Since DNS will always return ovsdbserver-xb.openstack.svc... this fields
+	// are an string instead of an array of strings. The add and removal of entries
+	// will be done using the DNSData CR, but this fields will remain the same on scale-up/down
+	dbAddress := ""
+
+	// DNSData is needed to provide DNS outside the cluster (edpm nodes) which needs to have
+	// spec.NetworkAttachment != ""
+	if instance.Spec.NetworkAttachment != "" {
+		for i, ovnPod := range podList.Items {
+			if i >= int(*(instance.Spec.Replicas)) {
+				Log.Info(fmt.Sprintf("In iteration %v is OOO, exit", i))
+				break
+			}
+			var dnsName string
+			var dnsIP string
+			var hostnames []string
+			dnsName = "dns-" + ovnPod.Name
+			// Get Hostname
+			svc, err := service.GetServiceWithName(
+				ctx,
+				helper,
+				ovnPod.Name,
+				ovnPod.Namespace,
+			)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			hostname := svc.ObjectMeta.Annotations[infranetworkv1.AnnotationHostnameKey]
+			hostnames = append(hostnames, hostname)
+
+			// Get IP
+			net_stat, err := nad.GetNetworkStatusFromAnnotation(ovnPod.Annotations)
+			if err != nil {
+				Log.Info(fmt.Sprintf("Error while getting the NAD for pod %s: %v", ovnPod.Name, err))
+				return ctrl.Result{}, err
+			}
+			for _, v := range net_stat {
+				if v.Interface == instance.Spec.NetworkAttachment {
+					// Return only IPv4 at the moment
+					if strings.Count(v.IPs[0], ":") == 0 {
+						dnsIP = v.IPs[0]
+						break
+					}
+				}
+			}
+
```

Comment:
shouldn't we somehow handle the case where dnsIP is empty here?

Reply:
I think we should continue the ovnPod loop if we don't find a valid IP. As right now it will create the DNS record without an IP (which not only will not work but difficult the debugging process) Will add a check and will print a message if there's no available IP.

Reply:
Done